### PR TITLE
Phishing Campaign - GetCampaignIncidentsInfo handling deleted incidents 

### PR DESCRIPTION
**Pull request by rshunim**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-6989)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py [hunk 1]
```diff
@@ -39,6 +39,12 @@ def update_incident_with_required_keys(incidents: List, required_keys: List):
         :param required_keys: keys need to be updated
 
     """
+    for incident in incidents:
+        res = demisto.executeCommand('GetIncidentsByQuery', {'query': "id:({})".format(incident['id'])})
```
Comment (dorschw) [inline Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py:43]:
```suggestion
        res = demisto.executeCommand("GetIncidentsByQuery", {"query": f"id:({incident['id']})"})
```
f-strings > `.format`
Please check manually

File: Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py [hunk 2]
```diff
@@ -39,6 +39,12 @@ def update_incident_with_required_keys(incidents: List, required_keys: List):
...
+        # If an incident is deleted, remove it from the list to avoid a key error
+        if res[0]['Contents'] == '[]':
```
Comment (dorschw) [inline Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py:45]: is this safe enough? (avoiding `IndexError`s)

File: Packs/Campaign/ReleaseNotes/3_2_24.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
+#### Scripts
+
+##### GetCampaignIncidentsInfo
+
+- Fixed an issue where ***GetCampaignIncidentsInfo*** was failed when some incidents were deleted.
```
Comment (dorschw) [inline Packs/Campaign/ReleaseNotes/3_2_24.md:6]:
```suggestion
- Fixed an issue causing the script to fail when incidents were deleted.
```

File: Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py (hunk 2 above)
Comment (rshunim) [reply Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py:45]: Fixed

File: Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py (hunk 1 above)
Comment (rshunim) [reply Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py:43]: Works

File: Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py [hunk 4]
```diff
@@ -39,6 +39,12 @@ def update_incident_with_required_keys(incidents: List, required_keys: List):
         :param required_keys: keys need to be updated
 
     """
+    for incident in incidents:
+        res = demisto.executeCommand("GetIncidentsByQuery", {"query": f"id:({incident['id']})"})
+        # If an incident is deleted, remove it from the list to avoid a key error
+        if res[0].get('Contents') == '[]':
+            incidents.remove(incident)
```
Comment (dorschw) [inline Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py:46]: Modifying a list when iterating on it is a bad practice, consider

```python
incidents = [incident for incident in incidents if demisto.executeCommand("GetIncidentsByQuery", {"query": f"id:({incident['id']})"})[0].get("Contents") != "[]"]
```
or the non-comprehension equivalent, which would be easier to debug. 

(sorry for not mentioning it before)

File: Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py [hunk 5]
```diff
@@ -39,6 +39,12 @@ def update_incident_with_required_keys(incidents: List, required_keys: List):
         :param required_keys: keys need to be updated
 
     """
+    for incident in incidents:
+        res = demisto.executeCommand("GetIncidentsByQuery", {"query": f"id:({incident['id']})"})
+        # If an incident is deleted, remove it from the list to avoid a key error
+        if res[0].get('Contents') == '[]':
```
Comment (dorschw) [inline Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py:45]: Please add a unit test for this case, mocking one deleted incident and one that isn't, making sure only the non-deleted one is returned.

**Review by dorschw — COMMENTED** [review]
Nice

File: Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py (hunk 4 above)
Comment (rshunim) [reply Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py:46]: You are absolutely right, this is an important point.

File: Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampainIncidentsInfo_test.py [hunk 6]
```diff
@@ -133,7 +134,7 @@ def test_some_error(mocker):
     # run
     try:
         main()
-        assert False, 'SystemExit should occurred'
+        pytest.fail(msg='SystemExit should occurred')
```
Comment (dorschw) [inline Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampainIncidentsInfo_test.py:137]: If you feel like being a good scout, let's change this to `with pytest.raises(SystemExit): main()`, it's the same but more explicit. Not urgent anyway.

File: Packs/Campaign/ReleaseNotes/3_3_1.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+
+#### Scripts
+
+##### GetCampaignIncidentsInfo
+
+- Fixed an issue causing the script to fail when incidents were deleted.
```
Comment (dorschw) [inline Packs/Campaign/ReleaseNotes/3_3_1.md:6]:
```suggestion
- Fixed an issue causing the script to fail after incidents are deleted.
```

Comment (dorschw) [reply Packs/Campaign/ReleaseNotes/3_3_1.md:6]: @ShirleyDenkberg Which of the options do you think is clearer?

**Review by dorschw — APPROVED** [review]
Good job!

**Pull request merged**